charts/consumer-app/templates/deployment.yaml

```diff
@@ -0,0 +1,254 @@
+          {{- end }}
+          {{- if and (hasKey .Values.kafka "labeledInputPatterns") (.Values.kafka.labeledInputPatterns) }}
+            - name: "{{ .Values.configurationEnvPrefix }}_LABELED_INPUT_PATTERNS"
+              value: "{{- range $key, $value := .Values.kafka.labeledInputPatterns }}{{ $key }}={{ $value }},{{- end }}"
+          {{- end }}
+          {{- if hasKey .Values.kafka "groupId" }}
+            - name: "{{ .Values.configurationEnvPrefix }}_GROUP_ID"
+              value: {{ .Values.kafka.groupId | quote }}
+          {{- end }}
+          {{- range $key, $value := .Values.secrets }}
+            - name: "{{ $key }}"
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "common-app.fullname" . }}
+                  key: "{{ $key }}"
+          {{- end }}
+          {{- range $key, $value := .Values.secretRefs }}
+            - name: "{{ $key }}"
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $value.name }}
+                  key: "{{ $value.key }}"
+          {{- end }}
+          {{- range $key, $value := .Values.commandLine }}
+            - name: "{{ $root.Values.configurationEnvPrefix }}_{{ $key }}"
+              value: {{ $value | quote }}
+          {{- end }}
+          {{- range $key, $value := .Values.env }}
+            - name: {{ $key | quote }}
+              value: {{ $value | quote }}
+          {{- end }}
+            - name: JAVA_TOOL_OPTIONS
+              value: '-Dcom.sun.management.jmxremote.port={{ .Values.jmx.port }}
+                -Dcom.sun.management.jmxremote.authenticate=false
+                -Dcom.sun.management.jmxremote.ssl=false
+                {{- if .Values.jmx.enabled }}
+                -Djava.rmi.server.hostname={{ .Values.jmx.host }}
+                -Dcom.sun.management.jmxremote.rmi.port={{ .Values.jmx.port }}
+                {{- end }}
+                -XX:MaxRAMPercentage={{ printf "%.1f" .Values.javaOptions.maxRAMPercentage }}
+                {{ .Values.javaOptions.others | join " " }}'
+        {{- if or (.Values.files) (and .Values.persistence.enabled .Values.statefulSet) (.Values.secretFilesRefs) }}
+          volumeMounts:
+            {{- range $key, $value := .Values.files }}
+            - name: config
+              mountPath: {{ printf "%s/%s" $value.mountPath $key | quote }}
+              subPath: {{ $key | quote }}
+            {{- end }}
+            {{- range .Values.secretFilesRefs }}
+            - name: {{ .volume }}
+              mountPath: {{ .mountPath }}
+              {{- if .readOnly }}
+              readOnly: true
+              {{- end }}
+              {{- if .subPath}}
+              subPath: {{.subPath }}
+              {{- end }}
+            {{- end }}
+            {{- if and .Values.persistence.enabled .Values.statefulSet }}
+            - name: datadir
+              mountPath: /tmp/kafka-consumer
```

Comment:
@raphala What is the purpose of this? For Kafka Streams there is a dedicated temp dir which is used for persistent state stores. However, here it seems unnecessary?